docs/tasks/administer-cluster/setup-ha-etcd-with-kubeadm.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Setup a Highly Available Etcd Cluster With Kubeadm
+---
+
+{% capture overview %}
+Kubeadm by default will create a self-hosted etcd cluster of one member. This is
+not a great default for a highly available control plane. This task walks
+through the process of creating a highly available etcd cluster of three members
+that can be used as an external etcd when using kubeadm to set up a kubernetes
+cluster.
+{% endcapture %}
+
+{% capture prerequisites %}
+* Three hosts that can talk to each other over ports 2379 and 2380. This
+  document assumes these default ports. However, they are configurable through
+  the kubeadm config file.
+* Each host must [have docker, kubelet, and kubeadm installed][toolbox].
+* SSH between hosts is required for certificate copying.
+
+[toolbox]: /docs/setup/independent/install-kubeadm/
+{% endcapture %}
+
+{% capture steps %}
+
+## Kubeadm Configuration File
+
+Create a kubeadm configuration file on each host that looks something like this:
+
+Pay attention to the `name` parameter in `extraArgs`. It needs to be unique per
+host. This example uses infra0, infra1, and infra2, but it can be any valid etcd
+member name just so long as the names all match in the `initial-cluster`
+section.
+
+```
+export HOST1=10.0.0.1 # Update HOST1, HOST2, and HOST3 with the IPs or resolvable names of your hosts
+export HOST2=10.0.0.2
+export HOST3=10.0.0.3
+export CURRENT_HOST="${HOST1}" # This must change for each host
+export NAME=infra0 # Update to use infra0, infra1, or infra2
+
+cat << EOF > kubeadmcfg.yaml
+apiVersion: "kubeadm.k8s.io/v1alpha1"
+kind: MasterConfiguration
+etcd:
+    serverCertSANs:
+    - "${CURRENT_HOST}"
+    peerCertSANs:
+    - "${CURRENT_HOST}"
+    extraArgs:
+        initial-cluster-token: etcd-cluster-1
```

Comment:
What is this used for?

Reply:
Ah, great catch. We don't even need this line as it has a default value and we are not spinning up multiple clusters. Here is the relevant documentation:

> If spinning up multiple clusters (or creating and destroying a single cluster) with same configuration for testing purpose, it is highly recommended that each cluster is given a unique initial-cluster-token. By doing this, etcd can generate unique cluster IDs and member IDs for the clusters even if they otherwise have the exact same configuration. This can protect etcd from cross-cluster-interaction, which might corrupt the clusters.